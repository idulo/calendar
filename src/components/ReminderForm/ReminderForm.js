import React from "react";
import moment from "moment";

import STYLES from "./ReminderForm.module.css";

class ReminderForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ""
    };
  }

  handleChange = event => {
    this.setState({ value: event.target.value });
    event.preventDefault();
  };

  handleSubmit = event => {
    this.setState({ value: "" });
    this.props.addReminder(this.state.value);
    event.preventDefault();
  };

  render() {
    return (
      <div className={STYLES.container}>
        <form onSubmit={this.handleSubmit}>
          <label>
            Create reminder for{" "}
            {moment()
              .month(this.props.selectedMonth)
              .date(this.props.selectedDay)
              .format("Do MMMM")}
            <input
              type="text"
              id="reminder"
              name="reminder"
              required
              value={this.state.value}
              placeholder="What is the reminder?"
              onChange={this.handleChange}
            />
          </label>
          <input
            className={STYLES.reminderForm}
            type="submit"
            value="Save reminder"
          />
        </form>
      </div>
    );
  }
}

export default ReminderForm;
