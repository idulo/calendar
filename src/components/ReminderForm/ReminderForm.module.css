.container {
  margin-bottom: 2rem;
}

.reminderForm {
  margin-top: 0.5rem;
}
