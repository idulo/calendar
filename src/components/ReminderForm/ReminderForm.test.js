import React from "react";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";
import ReminderForm from "./ReminderForm";

describe("ReminderForm", () => {
  const selectedDay = jest.fn().mockImplementation(3);
  const selectedMonth = jest.fn().mockImplementation(8);

  it("renders without crashing", () => {
    const div = document.createElement("div");

    ReactDOM.render(
      <ReminderForm selectedDay={selectedDay} selectedMonth={selectedMonth} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders correctly with selected day and month", () => {
    const Picker = create(
      <ReminderForm selectedDay={selectedDay} selectedMonth={selectedMonth} />
    );

    expect(Picker.toJSON()).toMatchSnapshot();
  });
});
