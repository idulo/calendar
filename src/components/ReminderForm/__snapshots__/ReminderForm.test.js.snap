// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReminderForm renders correctly with selected day and month 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Create reminder for
       
      NaNth undefined
      <input
        id="reminder"
        name="reminder"
        onChange={[Function]}
        placeholder="What is the reminder?"
        required={true}
        type="text"
        value=""
      />
    </label>
    <input
      className="reminderForm"
      type="submit"
      value="Save reminder"
    />
  </form>
</div>
`;
